pkg/liqo-controller-manager/namespaceMap-controller/namespacemap_controller.go

```diff
@@ -85,9 +83,28 @@ func (r *NamespaceMapReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
+// Events not filtered:
+// 1 -- the number of entries in DesiredMapping changes, so a namespace's request is added or removed.
+// 2 -- the NamespaceMap is deleted and has the NamespaceMapControllerFinalizer.
+func manageDesiredMappings() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If the NamespaceMap is deleted and has the NamespaceMapControllerFinalizer.
+			if !e.ObjectNew.GetDeletionTimestamp().IsZero() &&
+				ctrlutils.ContainsFinalizer(e.ObjectNew, namespaceMapControllerFinalizer) {
+				return true
+			}
+			// If an entry is added or removed in DesiredMapping
+			return !reflect.DeepEqual(e.ObjectOld.(*mapsv1alpha1.NamespaceMap).Spec.DesiredMapping,
+				e.ObjectNew.(*mapsv1alpha1.NamespaceMap).Spec.DesiredMapping)
+		},
+	}
+}
+
 // SetupWithManager monitors only updates on NamespaceMap.
 func (r *NamespaceMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&mapsv1alpha1.NamespaceMap{}).
+		WithEventFilter(manageDesiredMappings()).
```

Comment:
You can use `For(&mapsv1alpha1.NamespaceMap{}, builder.WithPredicates(p))`, it is the same thing but if one day this resource will have an owned resource, this filter will only be applied to the parent one

Reply:
thank you